server/compaction-coordinator/src/main/java/org/apache/accumulo/coordinator/CompactionFinalizer.java

```diff
@@ -99,15 +108,18 @@ public void commitCompaction(ExternalCompactionId ecid, KeyExtent extent, long f
     var ecfs =
         new ExternalCompactionFinalState(ecid, extent, FinalState.FINISHED, fileSize, fileEntries);
 
-    LOG.debug("Initiating commit for external compaction: {}", ecfs);
+    LOG.trace("Initiating commit for external compaction: {} {}", ecid, ecfs);
 
     // write metadata entry
-    context.getAmple().putExternalCompactionFinalStates(List.of(ecfs));
```

Comment:
It looks like the only other caller of this method is in `failCompactions` below. I'm wondering if we should use the same SharedBatchWriter for the failures and then remove `putExternalCompactionFinalStates` from the Ample interface. It's possible that someone could shutdown or kill a bunch of compactors (maybe via `accumulo-cluster prune`) and then the coordinator has a lot of failures to deal with.

Reply:
Removed the method in 22e5eec.   In that change also avoided creating an intermediate list of objects.  Would have needed to add more complex logic to the SharedBatchWriter to support  a batch of mutations and wanted to keep it simple.  So only used the shared batch writer when there was a single mutation, otherwise create a batch writer to handle multiple mutations.  This will make RPCs from compactors that report a failed compaction use the shared batch writer (this change makes all the RPCs use the shared batch writer, which is really good).  If dead compaction detection finds a lot of stuff to process, then it will run in the backgound w/ its own batch writer and not interfere w/ RPC related batch writes.